### DISCUSSION TITLE
Osiset compatible with CLI 3.0 for Post-checkout app extension

### DISCUSSION THREAD

**Discussion by sushilsth** [osiset/laravel-shopify#1160] [General]
Hi,

I have made the Shopify using this package and have already made LIVE in marketplace. Now, I want to implment theme app extension "Post Checkout UI". As per the requirement, I need to used Shopify CLI 3.0 [refernce link [reference link](https://shopify.dev/apps/checkout/post-purchase/getting-started-post-purchase-extension) ] where as Osiset donot have such directory.

I tried follow all the step mentioned in https://shopify.dev/apps/checkout/post-purchase/getting-started-post-purchase-extension but I am not able recognize web directory as mentioned in "Step 3: Add Shopify CLI as a project dependency >> 4".

Have anyone used Shopify CLI 3.0 in Osiset. Please help.

**Comment by sushilsth** [comment]
@osiset  any insights you have on this ?

**Comment by gosman** [comment]
Would also be interested in this

**Comment by Kyon147** [comment]
Much like any of the CLI options, you can just install them inside your current app directory.

Just run `shopify extension create post-purchase` for the post-purchase as an example. You need to have the `shopify cli` installed globally for that to work.

The docs are bad at just leaning towards the ruby/php cli scaffolded packages but you can create any extension inside any project.

I've recently used this package with the App Embed block extension fine, without the "CLI" installed in the project directly. 

`npm run scaffold extension` is just the cli walkthoguht to build an extension if you have the cli installed as part of the package itself.

**Reply by benson336** [reply]
@Kyon147 What's different between creating app extension by running `shopify extension create post-purchase` and by following this [guide](https://github.com/osiset/laravel-shopify/wiki/How-to-create-App-Extension)? Appreciate your help.

**Reply by Kyon147** [reply]
Hi @benson336 the guide is explaining how to install an extension when you have the Shopify cli installed locally per package. 

I personally just prefer to use a global installation of the Shopify cli that's all. The steps are almost identical.
